### PR TITLE
Support return of shard groups for time range

### PR DESCRIPTION
**Pull request by otoolep**
Metastore now supports return shard groups for a given database, retention policy, and time range.

### CODE REVIEW

File: meta/store_test.go [hunk 1]
```diff
@@ -389,6 +389,40 @@ func TestStore_CreateShardGroup(t *testing.T) {
...
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	} else if sgs[0].ID != 1 {
+		t.Fatalf("unexpected shard group: %#v", sgs[0])
+	}
+	if sgs, err := s.ShardGroupsByTimeRange("db0", "rp0", time.Date(2002, time.January, 1, 0, 0, 0, 0, time.UTC), time.Date(2002, time.January, 2, 0, 0, 0, 0, time.UTC)); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 0 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	}
 }
```
Comment (benbjohnson) [inline meta/store_test.go]: This test is getting kinda long. We don't need to separate it out yet but I'm worried it'll start getting too long and unweidly.

Comment (otoolep) [inline meta/store_test.go]: I agree. I will move it to its own test.

Comment (otoolep) [inline meta/store_test.go]: Well, let me ask you this. What is a clean way to test it, without, in effect, just retesting initial shard creation? There really isn't a way that I can see. That why I kinda just slammed it in here. ;-)

**Comment by jwilder** [comment]
:+1:

**Comment by otoolep** [comment]
@benbjohnson -- I split it out.

File: meta/store_test.go (hunk 1 above)
Comment (jwilder) [inline meta/store_test.go]: Maybe pull each of those if/else block into something like `assertShardGroupsInTimeRange(db,rp, time, 1)`, `assertShardGroupsInTimeRange(db,rp, time, 0)`, etc..?

Comment (otoolep) [inline meta/store_test.go]: Works for me @jwilder

**Comment by otoolep** [comment]
DRYed it out a bit, @jwilder and @benbjohnson

**Comment by otoolep** [comment]
OK, I presume this is OK, since @jwilder +1 the production code and it is unchanged. Let me know if you have any questions @benbjohnson

**Pull request merged**